cds/modules/deposit/indexer.py

```diff
@@ -43,3 +47,52 @@ def cdsdeposit_indexer_receiver(
     if record['$schema'] in [project_schema, video_schema]:
         json['_cds']['state'] = deposit['_cds']['state']
         json['_files'] = deposit['_files']
+
+
+class CDSRecordIndexer(RecordIndexer):
+
+    def index(self, deposit, action='commit'):
+        project_schema = current_jsonschemas.path_to_url(Project._schema)
+        if action == 'publish':
+            if deposit['$schema'] == project_schema:
+                # index videos (records)
+                pid_values = Project(data=deposit).video_ids
+                ids = [
+                    str(p.object_uuid)
+                    for p in PersistentIdentifier.query.filter(
+                        PersistentIdentifier.pid_value.in_(pid_values)).all()
+                ]
+                # index project (record)
+                _, record = deposit.fetch_published()
+                ids.append(str(record.id))
+                # index deposit (deposit)
+                # FIXME index also deposit videos?
+                ids.append(str(deposit.id))
+                super(CDSRecordIndexer, self).bulk_index(iter(ids))
+            else:
+                _, record = deposit.fetch_published()
+                super(CDSRecordIndexer, self).index(record)
+        elif action in ('edit', 'discard', 'commit'):
+            super(CDSRecordIndexer, self).index(deposit)
+        elif action == 'delete':
+            self.delete(deposit)
+
+    def delete(self, record):
+        video_schema = current_jsonschemas.path_to_url(Video._schema)
+        project_schema = current_jsonschemas.path_to_url(Project._schema)
+        if record['$schema'] == video_schema:
```

Comment:
Above you are not checking for both project and video, but only project and then `else`.
Any particular reason?

Reply:
Because you can publish a project or a video only...I could do the same below to be consistent if this is your point.